cabal.project

```diff
@@ -66,7 +89,9 @@ constraints:
   ghc-check -ghc-check-use-package-abis,
   ghc-lib-parser-ex -auto,
   stylish-haskell +ghc-lib,
-  fourmolu -fixity-th
+  fourmolu -fixity-th,
+  setup.happy == 1.20.1.1,
+  happy == 1.20.1.1
```

Comment:
I've just build this branch, and found that hiedb's use of `dfs` (in `HieDb.Query`) is incompatible with algebraic-graphs-0.7.

I successfully compiled this branch with the following command:

```
ghcup compile hls -g wip/9.6 --git-describe-version --ghc 9.6.0.20230210 --cabal-update -- --allow-newer -j4 --constraint="algebraic-graphs==0.6.1"
```

Perhaps, it might be good to add `algebraic-graphs == 0.6.1` here?